### ISSUE TITLE
FR: directory-structure ToC in CBZ

### ISSUE THREAD

**Issue by mergen3107** [koreader/koreader#7877]
**Does your feature request involve difficulty completing a task? Please describe.**
For CBZ files, could you please make an option to build ToC from directories inside CBZ? 
For example, many Manga archives look like this:
```
Volume 1.cbz:
- Chapter 1/
- - file1.png
- - file2.png
- - …
- Chapter 2/
- - file1.png
- - file2.png
- - …
- …
```

**Describe the solution you'd like**
An option to build ToC from directories. 

**Describe alternatives you've considered**
For now I use Kindle Comic Converter to convert cbz to epub. It works OK, but for each such epub I have to set `flat` rendering mode manually (because my default is `book` for non-Manga books), and enable publisher style. Otherwise a picture does not fit one page. A little tedious, but it works as an alternative (and has ToC).
I noticed that such epubs take longer to open (about 5-7 seconds for a 100 MB file) than their cbz variants (around 2-3 seconds). Is this expected?

Thank you!

**P. S.** Is there a way to zoom whole page in a cbz book? When I try to pinch and zoom it selects a panel. Sometimes there is fine-printed text in between panels which I cannot zoom to. Epubs have zoom though

**Comment by mergen3107** [comment]
@Frenzie
by Upstream you mean KCC?

**Comment by Frenzie** [comment]
KCC? 

I mean that in principle, the ToC comes from MuPDF.

**Comment by mergen3107** [comment]
Is this the right mupdf github page? https://github.com/ArtifexSoftware/mupdf

**Comment by Frenzie** [comment]
Yes, but that's just a mirror. They have their bug tracking at https://bugs.ghostscript.com/ and there's also an IRC channel somewhere.

**Comment by mergen3107** [comment]
Thank you!

**Comment by KaMyKaSii** [comment]
@mergen3107 did you report this in the mupdf bug tracker?

**Comment by mergen3107** [comment]
@KaMyKaSii 
No, I didn’t. I just gave up because most of the CBZ manga files are usually 200 pages or so, so ToC need is not that urgent :D

**Comment by KaMyKaSii** [comment]
I'm creating my cbz files with at least 25 chapters each so it's at least 800 pages per file. I just created an [issue](https://bugs.ghostscript.com/show_bug.cgi?id=705624) there, I hope I selected the right component.

**Comment by mergen3107** [comment]
@KaMyKaSii 
Has anyone replied yet? I think I can't see changes on that website until I log in

**Comment by KaMyKaSii** [comment]
> @KaMyKaSii Has anyone replied yet? I think I can't see changes on that website until I log in

No one yet

**Comment by mergen3107** [comment]
Fixed by #9804 
Thank you!!!
